Feature/navbar

mhmdwSamir opened the pull request:
check navbar and give me your feedback

mhmdwSamir committed:
generate SharedModule to wrap shared component ,
generate navbarComponent in shared Module ,
render navbar in app.component.html  file ,
start building NavBar

mhmdwSamir committed:
add fontawsome libarary to index.html file ,
use " Lato Font Awsome to Style.css file  " ,
finish styling of   layout of navbar bart

mhmdwSamir committed:
add hover effects to links element

mhmdwSamir committed:
add hover effects to icons on navbar

a7md940 commented on src/styles.scss:
```diff
@@ -1 +1,9 @@
 /* You can add global styles to this file, and also import other style files */
+
+@import url('https://fonts.googleapis.com/css2?family=Lato:wght@100;300;400;700;900&display=swap');
```
best practices to download the `font` and put it into `application assets folder`.

ahmedradwan94 replied:
best practices to download the `font` and put it into application `assets` folder.

a7md940 commented on src/index.html:
```diff
@@ -5,6 +5,7 @@
   <title>ToDoApp</title>
   <base href="/">
   <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/all.min.css" integrity="sha512-1ycn6IcaQQ40/MKBW2W4Rhis/DbILU74C1vSrLJxCq57o941Ym01SwNsOMqvEBFlcgUa6xLiPY/NS5R+E6ztJQ==" crossorigin="anonymous" referrerpolicy="no-referrer" />
```
Also here we want to download the `css file` and put it into `assets` folder.

ahmedradwan94 replied:
Also here we want to download the `css` file and put it into `assets` folder.

a7md940 commented on src/app/shared/shared-routing.module.ts:
```diff
@@ -0,0 +1,10 @@
+import { NgModule } from '@angular/core';
```
`SharedModule` must not have `RoutingModule`, because it's contains only utilities `Components, Directives and Pipes`.

ahmedradwan94 replied:
`SharedModule` must not have `RoutingModule`, because it's contains only utilities Components, Directives and Pipes.
So you have delete it.

a7md940 commented on src/app/shared/nav-bar/nav-bar.component.scss:
```diff
@@ -0,0 +1,73 @@
+.navBar{
```
css naming conversion doesn't use `CamelCase` but uses `Kebab case`.
`.navBar` is wrong, `.nav-bar` is right.

ahmedradwan94 replied:
css naming conversion doesn't use CamelCase but uses Kebab case.
.navBar is wrong, .nav-bar is right.

ahmedradwan94 commented on src/app/app.component.html:
```diff
@@ -1,520 +1,6 @@
-<!-- * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * -->
-
-<router-outlet></router-outlet>
+    <div class="blankBody"> blankBody </div>
```
Where is `router-outlet` element ?

ahmedradwan94 replied:
@mhmdwSamir  Did you put `router-outlet` in `app.component.html` file ?

mhmdwSamir replied:
Done  , 
as i didnt use navigation for right now so i missed it

ahmedradwan94 replied:
Ok

ahmedradwan94 commented on src/app/app.component.html:
```diff
@@ -1,520 +1,6 @@
-<!-- * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * -->
-
-<router-outlet></router-outlet>
+    <div class="blankBody"> blankBody </div>
```
Where is `router-outlet` element ?

mhmdwSamir committed:
make all requested view ,
download font awsome , font manulllay and link it ,  delete shared routing module ,
change name of navBar to nav-bar ,

ahmedradwan94 commented on src/app/app.component.scss:
```diff
@@ -5,14 +5,13 @@
     // max-height: 100vh; with it flexbasis dont work
   .backCover{
```
You should change all `css classes` to be `kebab case`

mhmdwSamir committed:
change classe name to applay kebab case

mhmdwSamir committed:
fix wrong spelling in class name

mhmdwSamir committed:
add router -outlet in app compoent file to start navigation corectly
